src/csm_testing/tests/sat_functional/bootprep/test_bootprep.py

```diff
@@ -0,0 +1,416 @@
+        """Test creating a CFS configuration with no layers"""
+        result = self.run_bootprep('no-layers-config.yaml', '--format json')
+
+        report = json.loads(result.stdout.decode())
+        self.assertEqual(1, len(report['configurations']))
+        self.assertEqual(f'{self.test_prefix}-no-layers', report['configurations'][0]['name'])
+
+        self.items_to_delete['configurations'].append(f'{self.test_prefix}-no-layers')
+
+    @skip_test_if_csm_var_missing(['version', 'commit_hash'])
+    def test_product_layers(self):
+        """Test creating multiple CFS configurations using product-based layers"""
+        result = self.run_bootprep('product-layers-config.yaml', '--format json')
+
+        report = json.loads(result.stdout.decode())
+        self.assertEqual(5, len(report['configurations']))
+
+        # Query CFS for the configurations and ensure they exist
+        # Validate certain elements of the created CFS configurations including:
+        #   - configuration names
+        #   - clone URLs
+        #   - commit hashes
+
+        for config in report['configurations']:
+            self.items_to_delete['configurations'].append(config['name'])
+
+    def test_git_layers(self):
+        """Test creating multiple CFS configurations with git-based layers"""
+        result = self.run_bootprep('git-layers-config.yaml', '--format json')
+
+        report = json.loads(result.stdout.decode())
+        self.assertEqual(3, len(report['configurations']))
+
+        # Query CFS for the configurations and ensure they exist
+        # Validate certain elements of the created CFS configurations including:
+        #   - configuration names
+        #   - clone URLs
+        #   - commit hashes
+
+        for config in report['configurations']:
+            self.items_to_delete['configurations'].append(config['name'])
+
+    def test_special_parameters(self):
+        """Test creating a CFS configuration with special parameters"""
+        result = self.run_bootprep('special-parameters-config.yaml', '--format json')
+
+        report = json.loads(result.stdout.decode())
+        self.assertEqual(1, len(report['configurations']))
+        self.assertEqual(f'{self.test_prefix}-special-parameters',
+                         report['configurations'][0]['name'])
+
+        # Query CFS for the configurations and ensure they exist
+        # Validate certain elements of the created CFS configurations including:
+        #   - configuration names
+        #   - special parameters values
+
+        for config in report['configurations']:
+            self.items_to_delete['configurations'].append(config['name'])
+
+    def test_additional_inventory(self):
+        """Test creating CFS configurations with additional inventory"""
+        result = self.run_bootprep('additional-inventory-config.yaml', '--format json')
+
+        report = json.loads(result.stdout.decode())
+        self.assertEqual(1, len(report['configurations']))
```

Comment:
Since you defined two configurations in additional-inventory-config.yaml file, It should be equal to 2.
```suggestion
        self.assertEqual(2, len(report['configurations']))
```